src/components/pages/PageDecksEdit/PageDecksEdit.jsx

```diff
@@ -0,0 +1,237 @@
+  const [selectedConflictIds, setSelectedConflictIds] = useState([]);
+  const [deck, setDeck] = useState({});
+
+  const getCardOptions = useCallback(
+    async (cardType) => {
+      const conflictCards = await firebaseCollectionsHelper
+        .getAllCardsDataByType(user.uid, cardType);
+
+      return conflictCards.map((card) => ({
+        label: card.cardTitle,
+        value: card.id,
+      }));
+    }, [user.uid],
+  );
+
+  const populateCharacters = useCallback(
+    async () => {
+      const newCharacterOptions = await getCardOptions('Character');
+      setCharacterOptions(newCharacterOptions);
+
+      return newCharacterOptions;
+    },
+    [getCardOptions],
+  );
+
+  const populateCircumstances = useCallback(
+    async () => {
+      const newCircumstanceOptions = await getCardOptions('Circumstance');
+      setCircumstanceOptions(newCircumstanceOptions);
+
+      return newCircumstanceOptions;
+    },
+    [getCardOptions],
+  );
+
+  const populateConflicts = useCallback(
+    async () => {
+      const newConflictOptions = await getCardOptions('Conflict');
+      setConflictOptions(newConflictOptions);
+
+      return newConflictOptions;
+    },
+    [getCardOptions],
+  );
+
+  const populateData = useCallback(
+    async () => {
+      const characters = await populateCharacters();
+      const circumstances = await populateCircumstances();
+      const conflicts = await populateConflicts();
+
+      await Promise.all([characters, circumstances, conflicts])
+        .then((values) => values);
+
+      const initialDeckData = await firebaseCollectionsHelper
+        .getDeckData(id);
+      setDeck(initialDeckData);
+
+      const selectedCharacters = initialDeckData.characterCards
+        .map((card) => characters.find((character) => (
+          character.value === card.cardRef.id
+        )))
+        .filter((i) => i);
+      setSelectedCharacterIds(selectedCharacters);
+
+      const selectedCircumstances = initialDeckData.circumstanceCards
+        .map((card) => circumstances.find((circumstance) => (
+          circumstance.value === card.cardRef.id
+        )))
+        .filter((i) => i);
+      setSelectedCircumstanceIds(selectedCircumstances);
+
+      const selectedConflicts = initialDeckData.conflictCards
+        .map((card) => conflicts.find((conflict) => (
+          conflict.value === card.cardRef.id
+        )))
+        .filter((i) => i);
+      setSelectedConflictIds(selectedConflicts);
+
+      setIsLoading(true);
+    },
+    [id, populateCharacters, populateCircumstances, populateConflicts],
+  );
+
+  function changeDeck(key, value) {
+    const newDeck = { ...deck };
+    newDeck[key] = value;
+    setDeck(newDeck);
+  }
+
+  const handleUpdate = useCallback(
+    async () => {
+      const newDeck = {
+        ...deck,
+        name: deck.name,
+        description: deck.description,
+        characterCards: selectedCharacterIds.map((cardId) => (
```

Comment:
You're repeating the cards object three times combine it into a function to produce the data